components/Hero.tsx

```diff
@@ -3,11 +3,25 @@
 import { Button } from 'shared'
 import { useLang, useLocalizedRoutes, useTranslations } from 'hooks'
 import Image from 'next/image'
+import { useAuthContext } from 'providers/AuthProvider'
+import { useProgressContext } from 'providers/ProgressProvider'
+import { keys } from 'lib/progress'
 
 export default function Hero() {
   const routes = useLocalizedRoutes()
   const lang = useLang()
   const t = useTranslations(lang)
+  const { account } = useAuthContext()
+  const { progress } = useProgressContext()
+
+  const progressHrefFragement =
+    progress === keys[keys.length - 1]
```

Comment:
This logic works correctly for now since the last element in the keys array is currently equal to 'CH2OUT1'.
However, this line of code will not work properly when we add further chapters.

For example,
```
progress = 'CH2OUT1'
keys[keys.length - 1] = 'CH3OUT1'
```

So `progressHrefFragement` according to this logic will be `2`. Whereas it should be `3`.

A more robust approach would be to check if the user progress is equal to last lesson of current chapter. The following code achieves that.

```suggestion
    progress === keys.filter(ele => ele.includes(progress.substring(0,3))).pop()
```

Reply:
So this ternary function is supposed to produce two results
1. When the user is at the very end of the game, i.e. their progress === keys.[keys.length -1] they are sent to the chapter Id +1 so that they can see that the next lesson is 'coming soon'
1. All other progress will direct you to the chapter based on your current progress

Reply:
Okay, this makes sense.

So what I understand is that,
Say I am on `CH2OUT1`, and instead of pressing next, I go back to the home screen.
So when I reopen chapter selection, I will be directed to _Chapter-2_, instead of _Chapter-3_.

Reply:
Correct. As I said in my comment in your PR #477 I like the idea of automatically being able to skip the end screen progress, but we need to add that logic where the progress is tracked and not through any sub-logic that feeds from the progress as It will inevitably cause some headache down the line

Reply:
With my progress at `CH1OUT2` I will still need to progress forward before the next chapter is unlocked, we can add some logic so that the outro screen progress is automatically skipped but we also need to consider in the first chapter we are expecting/requiring the user to log in so this logic would need to check if the user is logged in first

Reply:
Got it!
So skipping the outro page will be unreasonable since we want the user to signup up by the outro of Chapter-1.

Reply:
So logic-wise, the code is great!
So we can separate these changes into a `common` function, and we shall be good to go!